rules.mk

```diff
@@ -0,0 +1,32 @@
+#
+# Common rules for makefiles for the PX4 bootloaders
+#
+
+BUILD_DIR	 = build_$(TARGET_FILE_NAME)
+
+OBJS		:= $(addprefix $(BUILD_DIR)/, $(patsubst %.c,%.o,$(SRCS)))
+#OBJS		:= $(patsubst %.c,%.o,$(SRCS))
+DEPS		:= $(OBJS:.o=.d)
+
+ELF		 = $(BUILD_DIR)/$(TARGET_FILE_NAME).elf
+BINARY		 = $(BUILD_DIR)/$(TARGET_FILE_NAME).bin
+
+all:		$(BUILD_DIR) $(ELF) $(BINARY)
+
+# Compile and generate dependency files
+$(BUILD_DIR)/%.o:	%.c
```

Comment:
@davids5 : Is there a more elegant way to handle the dependency on the particular source file so that you don't have to specifically mention its file extension?